### ISSUE TITLE
Backend for configurable session-management

### ISSUE THREAD

**Issue by chandel-aman** [PalisadoesFoundation/talawa-api#1637]
**Is your feature request related to a problem? Please describe.**

Currently, the timeout for Talawa-Admin is hard-coded in Talawa-Api. However, there is no feature to set a custom timeout for different organizations.

**Describe the solution you'd like**

Implement configurable session timeout for each organization and provide admins and super-admins the ability to access and change the session timeout for an organization, with the default timeout set to 30 minutes.

**Describe alternatives you've considered**
NA

**Additional context**

Related to the issue in Talawa-Admin [#1305](https://github.com/PalisadoesFoundation/talawa-admin/issues/1305).

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by Cioppolo14** [comment]
@chandel-aman This is now assigned to you.  Please wait to submit PRs until you are assigned an issue. Also, take note of @palisadoes comment in the PR you submitted.

**Comment by palisadoes** [comment]
This would be much better served at the Community level with a single value for all organizations. Therefore it would be managed by the SuperAdmin using this profile page for the configuration
- https://github.com/PalisadoesFoundation/talawa-admin/issues/1334#issuecomment-1879522861


This must be implemented as a single value for all organizations

**Comment by chandel-aman** [comment]
> This would be much better served at the Community level with a single value for all organizations. Therefore it would be managed by the SuperAdmin using this profile page for the configuration
> 
>     * [Add the ability to change pre-login imagery talawa-admin#1334 (comment)](https://github.com/PalisadoesFoundation/talawa-admin/issues/1334#issuecomment-1879522861)
> 
> 
> This must be implemented as a single value for all organizations

Correct me if I'm wrong.

1. You're proposing that the session timeout be configurable at the community level, implying that organizations would fall under the community umbrella, and the session timeout data should be applicable universally to all the organizations under that community.
2. Additionally, you're recommending that only super-admins should have the authority to update the session timeout, excluding admins from this privilege.

**Comment by palisadoes** [comment]
yes

**Comment by xoldyckk** [comment]
@palisadoes what is the need for session timeouts per organization?
 
Is talawa application meant to be similar to slack in that for each organization there has to be a seperate customizable authentication flow? Users first create a talawa account to gain access to the application and organizations. Then they would have to sign in to each organization seperately and follow seperate authentication flows specific to those organizations.

Or is it meant to be similar to how discord communities are managed? Users sign in once, get a global session, get to choose which organization they want to join. When the session expires they have to sign in to the talawa application again. The only difference being that here only talawa admins would be able to create organizations unlike discord where normal users can make their own discord communities.

Superadmin is an ambiguous term that's why I've used the (talawa) admin term here. 

user roles:- ADMIN/USER
organization member roles:- MODERATOR/MEMBER

**Comment by palisadoes** [comment]
It's like discord. Only admins can create orgs, with global session management across orgs.

**Comment by palisadoes** [comment]
Unassigning due to inactivity

**Comment by lakshz** [comment]
I would like to work on this.

**Comment by Olatade** [comment]
Our policy is to assign no more than one issue to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.

**Comment by chandel-aman** [comment]
@palisadoes
I understand that I was unassigned from this issue due to inactivity, and I apologize for not being able to complete it earlier. I had some health issues that prevented me from working. However, I would like to take ownership of this issue again as I have already implemented it. Could you please reassign it to me? Thank you for your understanding.

**Comment by jyotiv737** [comment]
@chandel-aman Are you working on this issue?

**Comment by chandel-aman** [comment]
> @chandel-aman Are you working on this issue?

Yes, I have raised the [PR](https://github.com/PalisadoesFoundation/talawa-api/pull/1962) for this.

**Comment by palisadoes** [comment]
Unassigning due to no activity in the PR

**Comment by Ramneet04** [comment]
Hey I want to work on it. Can you assign it to me?
@palisadoes
